src/stateless_validation.cpp

```diff
@@ -1,17 +1,128 @@
 #include <algorithm>
+#include <bitset>
 #include <string>
 #include <vector>
 
 #include "veriblock/arith_uint256.hpp"
+#include "veriblock/blob.hpp"
 #include "veriblock/consts.hpp"
 #include "veriblock/stateless_validation.hpp"
 #include "veriblock/strutil.hpp"
 
+namespace {
+
+int getIntFromBits(const std::vector<bool>& bits,
+                   const uint32_t& fromIndex,
+                   const uint32_t& toIndex) {
+  int value = 0;
+  for (size_t j = fromIndex; j < toIndex; ++j) {
+    if (bits[j]) {
+      value += 1 << (j - fromIndex);
+    }
+  }
+  return value;
+}
+
+};  // namespace
+
 namespace VeriBlock {
 
 static const ArithUint256 MAXIMUM_DIFFICULTY(
     "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF");
 
+bool containsSplit(const std::vector<uint8_t>& pop_data,
+                   const std::vector<uint8_t>& btcTx_data) {
+  static const std::vector<uint8_t> magicBytes = {0x92, 0x7a, 0x59};
+
+  ReadStream buffer(btcTx_data);
+
+  try {
+    size_t lastPos = 0;
+    while (buffer.remaining() > 5) {
+      if (buffer.readBE<uint8_t>() != magicBytes[0] ||
+          buffer.readBE<uint8_t>() != magicBytes[1] ||
+          buffer.readBE<uint8_t>() != magicBytes[2]) {
+        continue;
+      }
+
+      lastPos = buffer.position();
+      // Parse the first byte to get the number of chunks, their positions and
+      // lengths
+      uint8_t descriptor = buffer.readBE<uint8_t>();
+
+      uint32_t chunks = 0;
+      uint32_t offsetLength = 4;
+      uint32_t sectionLength = 4;
+      for (int i = 0; i < 8; ++i) {
+        if ((descriptor >> i) & 1) {
+          if (i < 2) {
+            sectionLength += 1 << i;
+          } else if (i < 4) {
+            offsetLength += 1 << i;
+          } else {
+            chunks += 1 << (i - 4);
+          }
+        }
+      }
+
+      // Parse the actual chunk descriptors now that we know the sizes
+
+      uint32_t chunkDescriptorBitLength =
+          (chunks * offsetLength) + (sectionLength * (chunks - 1));
+      uint32_t chunkDescriptorBytesLength =
+          (chunkDescriptorBitLength + 8 - (chunkDescriptorBitLength % 8)) / 8;
+      uint32_t waste =
+          chunkDescriptorBytesLength * 8 - chunkDescriptorBitLength;
+
+      std::vector<uint8_t> chunkDescriptorBytes =
+          buffer.readSlice(chunkDescriptorBytesLength).reverse();
+      std::vector<bool> chunkDescriptor;
+      // Read from Slice bits
+      uint32_t chunkDescriptorLength = 0;
+      for (uint32_t i1 = 0; i1 < chunkDescriptorBytes.size(); ++i1) {
```

Comment:
for(const auto i1 : chunkDescriptorBytes)

Reply:
fixed